nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -79,11 +76,15 @@ LifecycleManager::LifecycleManager()
   transition_label_map_[Transition::TRANSITION_UNCONFIGURED_SHUTDOWN] =
     std::string("Shutting down ");
 
-  createLifecycleServiceClients();
-
-  if (autostart_) {
-    startup();
-  }
+  init_timer_ = this->create_wall_timer(
+    std::chrono::nanoseconds(10),
```

Comment:
Is 10 nanoseconds enough to make sure you got out of the constructor reliably? (It might be, I really don't know)

Reply:
The node is mono-threaded (no callback group for the timer), so the timer callback will not be executed before the end of the constructor.